parser.hh

```diff
@@ -290,7 +311,7 @@ struct AsciiFileInput : public Input
 	/**
 	 * Construct a parser input from a specified file descriptor.
 	 */
-	AsciiFileInput(int file);
+	AsciiFileInput(const std::string& name, int file);
```

Comment:
Please can you change the order of this (and the other subclass constructors) so that we can have a default `name = ""`.